### PR TITLE
Added optional double down action to blackjack

### PR DESCRIPTION
**Pull request by chisness**
Extends the game from just hit/stick to the 3rd action of doubling down, which means doubling the bet and getting only 1 additional card

### CODE REVIEW

File: gym/envs/toy_text/blackjack.py [hunk 1]
```diff
@@ -81,6 +84,8 @@ def __init__(self, natural=False):
         # Flag to payout 1.5 on a "natural" blackjack win, like casino rules
         # Ref: http://www.bicyclecards.com/how-to-play/blackjack/
         self.natural = natural
+        #Flag for allowing doubling down
```
Comment (pzhokhov) [inline gym/envs/toy_text/blackjack.py]: this comment should rather be in the docstring (could you also describe what natural stands for while at it then ^_^)

File: gym/envs/toy_text/blackjack.py [hunk 2]
```diff
@@ -90,15 +95,25 @@ def seed(self, seed=None):
 
     def step(self, action):
         assert self.action_space.contains(action)
-        if action:  # hit: add a card to players hand and return
+        if self.double_down:
```
Comment (pzhokhov) [inline gym/envs/toy_text/blackjack.py]: minor nitpick - given the `assert self.action_space.contains(action)` above, if action == 2, then self.double_down should always be true, so the outer if statement is unnecessary here

Comment (chisness) [reply gym/envs/toy_text/blackjack.py]: Thanks, fixed in new commit

File: gym/envs/toy_text/blackjack.py (hunk 1 above)
Comment (chisness) [reply gym/envs/toy_text/blackjack.py]: Thanks, updated in new commit

**Comment by chisness** [comment]
@pzhokhov I made those changes and a couple of other small ones in the new commit. I showed a test of 100 double_down actions to show the reward is always {-2, 0, 2} (except the case of natural Blackjack when natural=True, which gives an immediate reward of 1.5).

**Comment by christopherhesse** [comment]
Do we really want to extend these built-in environments instead of having people make their own environments outside the gym repo @pzhokhov?

**Comment by jkterry1** [comment]
Closing per #2259

**Comment by jkterry1** [comment]
That was a typo sorry, I'm closing this after discussion with @cpnota because this is intended to match the simplified blackjack game in Barto and Sutton

**Pull request closed without merging**